### PR TITLE
Add first pass on step services API docs

### PR DESCRIPTION
**Pull request by mozbhearsum**
A few of us met yesterday to talk about what the required HTTP API that step services need to implement would be. This is an attempt to start documenting it. I don't think we've fully decided what it will look like, but this is a start.

Here's a dump from the notes we took yesterday, with some open questions:
* Resource based (supports CRUD operations)
* https://github.com/lundjordan/services/blob/shipit-taskcluster/src/shipit_taskcluster/shipit_taskcluster/api.yml
* All steps support a base interface, some steps may have step-specific interfaces
* need to be careful about allowing step status to be overridden, otherwise we may violate signoff requirements
* pipeline will not be changing step status
* may need additional information returned with some statuses (eg: required/completed signoffs for signoffs steps)
  * or, use a step-specific endpoint for this sort of thing, and leave the main step api alone
* strong case to be made for allowing users to specify uids for pipelines (they can use names like Firefox-53.0-build1)
  * not so strong with steps - too many possible little variations in them, names not as intelligible to humans

Obviously this needs some more updates and fleshing out! What do you think of this so far, @lundjordan?

### CODE REVIEW

File: src/releng_docs/services_shipit.rst [hunk 1]
```diff
@@ -57,3 +60,44 @@ It does the following tasks on every run:
 
 
 .. _libmozdata: https://github.com/mozilla/libmozdata/
```
Comment (lundjordan) [inline src/releng_docs/services_shipit.rst]: should this go to the end of the rst file?

File: src/releng_docs/services_shipit.rst [hunk 2]
```diff
@@ -57,3 +60,44 @@ It does the following tasks on every run:
...
+* /
+** GET - Returns all steps with status (TODO: probably need pagination and filtering here)
+* /{uid}
+** PUT - Create a new Step
+** DELETE - Remove the given Step
+* /{uid}/definition
+** GET Returns the definition of the given step
+* /{uid}/status
+** GET Returns the status of the given step.
+*** Currently, one of: starting, running, stopping, exception, completed, failed
```
Comment (lundjordan) [inline src/releng_docs/services_shipit.rst]: I'd like to figure out the possible statuses. I'm not sure what we really gain from having [starting, running, stopping] vs just [running]. Particularly if we have a way to retrieve custom  service specific data summarizing state

As per last week's meeting, I was thinking we could try a minimal set of 4 statuses similar to aws step functions: in_progress, success, failure, cancelled

File: src/releng_docs/services_shipit.rst (hunk 1 above)
Comment (mozbhearsum) [inline src/releng_docs/services_shipit.rst]: Maybe? I assumed it was associated with shipit_bot_uplift. Maybe @La0 can tell us for sure?

Comment (La0) [inline src/releng_docs/services_shipit.rst]: Yes this library from Release Management is used to do the patch analysis in shipit_bot_uplift.
You can see the link used on https://docs.mozilla-releng.net/services_shipit.html#src-shipit-bot-uplift

File: src/releng_docs/services_shipit.rst (hunk 2 above)
Comment (mozbhearsum) [inline src/releng_docs/services_shipit.rst]: WFM! We can always adjust later.

File: src/releng_docs/services_shipit.rst (hunk 1 above)
Comment (mozbhearsum) [inline src/releng_docs/services_shipit.rst]: In that case, I think this link belongs here, not at the bottom of the file.

**Pull request merged**